ui/slider.js

```diff
@@ -157,6 +157,48 @@ return $.widget( "ui.slider", $.ui.mouse, {
 		}
 	},
 
+	_cleanValues: function( index, newValue ) {
+		var values,
+			i = 0 ;
+
+		if ( $.isArray( arguments[ 0 ] ) ) {
+			values = arguments[ 0 ] || this.options.values;
+			if ( values ) {
+				for ( ; i < values.length; i++ ){
+					if ( values[ i ] < this._valueMin() ) {
+						values[ i ] = this._valueMin();
+					}
+
+					if ( values[ i ]  > this._valueMax() ) {
+						values[ i ] = this._valueMax();
+					}
+				}
+				if ( values[ 0 ] > values[ 1 ] ) {
+					values[ 1 ] = values[ 0 ];
+				}
+				return values;
+			}
+		}
+
+		if ( arguments.length > 1 ) {
+			values = this.options.values;
+			if ( index === 0 && newValue < this._valueMin() ){
+				newValue = this._valueMin();
+			}else if ( index === 1 && newValue > this._valueMax() ){
+				newValue = this._valueMax();
+			}
+
+			if ( this.options.range === true && values.length === 2 ) {
+				if ( index === 0 && newValue > values[ index + 1 ] ) {
+					newValue = values[ index + 1 ];
+				}else if ( index === 1 && newValue < values[ index - 1 ] ) {
+					newValue = values[ index - 1 ];
+				}
```

Comment:
Why the `index +/- 1` logic? You already have the indices hard-coded anyway out of necessity.